File: .claude/skills/webapp-testing/test_partner_smoke.py
"""
Smoke test for partner pages
"""
from playwright.sync_api import sync_playwright

BASE_URL = 'http://localhost:3002'
PAGES = [
    ('/partner/login', '파트너 로그인'),
    ('/partner/find-account', '파트너 계정찾기'),
    ('/partner/mypage', '파트너 마이페이지'),
    ('/partner/mypage/edit', '파트너 내정보 수정'),
    ('/partner/mypage/profile', '파트너 프로필'),
    ('/partner/campaign_management/scheduled', '캠페인관리 예정'),
    ('/partner/campaign_management/progress', '캠페인관리 진행'),
    ('/partner/notification', '파트너 알림'),
]
REACT_ERRORS = ['Unhandled Runtime Error','Application error','TypeError:','ReferenceError:','Cannot read properties']

def run():
    results = []
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        ctx = browser.new_context()
        pg = ctx.new_page()
        pg.goto(BASE_URL)
        pg.wait_for_load_state('networkidle')
        pg.evaluate("""() => {
            localStorage.setItem('reviewx_auth_user', JSON.stringify({id:'partner_test_001',email:'test@partner.com',name:'테스트',role:'partner',business_name:'테스트주식회사'}));
            localStorage.setItem('reviewx_auth_role', 'partner');
            localStorage.setItem('reviewx_auth_token', 'mock_token');
        }""")
        pg.close()

        for path, label in PAGES:
            url = BASE_URL + path
            console_errs = []
            page = ctx.new_page()
            page.on('console', lambda m: console_errs.append(m.text) if m.type == 'error' else None)
            try:
                resp = page.goto(url, wait_until='domcontentloaded', timeout=15000)
                page.wait_for_load_state('networkidle', timeout=10000)
                status = resp.status if resp else 0
                html = page.content()
                rerrs = [e for e in REACT_ERRORS if e.lower() in html.lower()]
                jerrs = [e for e in console_errs if any(x.lower() in e.lower() for x in REACT_ERRORS)]
                ok = status == 200 and not rerrs and not jerrs
                results.append({'label':label,'status':status,'ok':ok,'rerrs':rerrs,'jerrs':jerrs[:2]})
            except Exception as e:
                results.append({'label':label,'status':0,'ok':False,'rerrs':[],'jerrs':[str(e)[:80]]})
            finally:
                page.close()
        browser.close()

    print('
' + '='*55)
    print('  파트너 페이지 스모크 테스트')
    print('='*55)
    for r in results:
        icon = '✅' if r['ok'] else '❌'
        print(f"{icon} [{r['status']}] {r['label']}")
        if r['rerrs']: print(f"    React: {r['rerrs']}")
        if r['jerrs']: print(f"    JS: {r['jerrs']}")
    passed = sum(1 for r in results if r['ok'])
    print('='*55)
    print(f'결과: {passed}/{len(results)} 통과')
    print('='*55)
    return passed == len(results)

if __name__ == '__main__':
    ok = run()
    exit(0 if ok else 1)

File: .claude/skills/webapp-testing/test_partner_smoke.mjs
import { chromium } from 'playwright';

const BASE_URL = 'http://localhost:3002';
const PAGES = [
  ['/partner/login', '파트너 로그인'],
  ['/partner/find-account', '파트너 계정찾기'],
  ['/partner/mypage', '파트너 마이페이지'],
  ['/partner/mypage/edit', '파트너 내정보 수정'],
  ['/partner/mypage/profile', '파트너 프로필'],
  ['/partner/campaign_management/scheduled', '캘페인관리 예정'],
  ['/partner/campaign_management/progress', '캘페인관리 진행'],
  ['/partner/notification', '파트너 알림'],
];
const REACT_ERR_PATTERNS = ['Unhandled Runtime Error','Application error','TypeError:','ReferenceError:','Cannot read properties'];

const browser = await chromium.launch({ headless: true });
const ctx = await browser.newContext();

const authPage = await ctx.newPage();
await authPage.goto(BASE_URL);
await authPage.waitForLoadState('networkidle');
await authPage.evaluate(() => {
  localStorage.setItem('reviewx_auth_user', JSON.stringify({
    id: 'partner_test_001', email: 'test@partner.com',
    name: '테스트파트너', role: 'partner', business_name: '테스트주식회사'
  }));
  localStorage.setItem('reviewx_auth_role', 'partner');
  localStorage.setItem('reviewx_auth_token', 'mock_token_partner');
});
await authPage.close();

const results = [];
for (const [path, label] of PAGES) {
  const url = BASE_URL + path;
  const consoleErrs = [];
  const page = await ctx.newPage();
  page.on('console', msg => { if (msg.type() === 'error') consoleErrs.push(msg.text()); });
  try {
    const resp = await page.goto(url, { waitUntil: 'domcontentloaded', timeout: 15000 });
    await page.waitForLoadState('networkidle', { timeout: 10000 });
    const status = resp ? resp.status() : 0;
    const html = await page.content();
    const reactErrs = REACT_ERR_PATTERNS.filter(p => html.toLowerCase().includes(p.toLowerCase()));
    const jsErrs = consoleErrs.filter(e => REACT_ERR_PATTERNS.some(p => e.toLowerCase().includes(p.toLowerCase())));
    const ok = status === 200 && !reactErrs.length && !jsErrs.length;
    results.push({ label, status, ok, reactErrs, jsErrs: jsErrs.slice(0, 2) });
  } catch (e) {
    results.push({ label, status: 0, ok: false, reactErrs: [], jsErrs: [String(e).slice(0, 80)] });
  } finally {
    await page.close();
  }
}
await browser.close();

console.log(String.fromCharCode(10) + "=".repeat(55));
console.log('  파트너 페이지 스모크 테스트');
console.log('='.repeat(55));
for (const r of results) {
  const icon = r.ok ? 'PASS' : 'FAIL';
  console.log(icon + ' [' + r.status + '] ' + r.label);
  if (r.reactErrs.length) console.log('    React: ' + r.reactErrs.join(', '));
  if (r.jsErrs.length) console.log('    JS: ' + r.jsErrs.join(', '));
}
const passed = results.filter(r => r.ok).length;
console.log('='.repeat(55));
console.log('결과: ' + passed + '/' + results.length + ' 통과');
console.log('='.repeat(55));
process.exit(passed === results.length ? 0 : 1);